pyuvdata/uvbeam.py

```diff
@@ -978,6 +978,76 @@ def __iadd__(self, other):
         self.__add__(other, inplace=True)
         return self
 
+    def get_beam_area(self, stokes='pseudo_I'):
+        """
+        Computes the integral of the beam, which has units of steradians
+
+        Currently, only the "pseudo Stokes I" beam is supported.
+        See Equations 4 and 5 of Moore et al. (2017) ApJ 836, 154
+        or arxiv:1502.05072 for details.
+        """
+        if self.beam_type is not 'power':
+            raise ValueError('beam_type must be power')
+        if self._data_normalization.value != 'peak':
+            raise ValueError('beam must be peak normalized')
+        if self.pixel_coordinate_system is not 'healpix':
+            raise ValueError('Currently only healpix format supported')
+        if self.data_array.shape[0] > 1:
+            raise ValueError('Expect scalar for power beam, found vector')
+
+        npix = self.data_array.shape[-1]
+        pol_array = self.polarization_array
+        if stokes == 'pseudo_I':
+            if 1 in pol_array:
+                stokes_I_ind = np.where(np.isin(pol_array, 1))[0][0]
+                beam = self.data_array[0, 0, stokes_I_ind]
+            elif -5 in pol_array and -6 in pol_array:
+                # Assume A_I = (B_xx + B_yy)/2
+                xx_ind = np.where(np.isin(pol_array, -5))[0][0]
+                yy_ind = np.where(np.isin(pol_array, -6))[0][0]
+                beam = 0.5 * (self.data_array[0, 0, xx_ind] + self.data_array[0, 0, yy_ind])
+            else:
+                raise ValueError('Do not have the right polarization information')
+        else:
+            raise NotImplementedError()
+
+        return np.sum(beam, axis=-1) * 4. * np.pi / npix
```

Comment:
I'm still worried that this command requires that the beam covers the full sky, and I'm actually not quite sure why you're dividing by npix (this is a sum not an average, right?). But if you want the area covered by the beam, you can use `healpy.pixelfunc.nside2pixarea` to get the pixel area and then multiply by self.Npixels.

Reply:
But naively I would've thought that if I'm only accepting Healpix, then the beams are by definition full sky?

I'm multiplying by 4pi/npix because it's meant to approximate an integral, rather than a sum. So the 4pi/npix is meant to be the dOmega differential element.

Reply:
Ah, thanks for pointing out this problem! I hadn't realized that you had built support for a non-full sky collection of Healpix pixels.

I've made the changes, working out the dOmega for nside in the object but using the healpy function in the tests to minimize the chances that I messed it up.

Reply:
That's perfect, thanks! I'm going to add one more little test to cover the case where we have a stokes I polarization for completeness and then I think we can merge it.